rskj-core/src/main/java/co/rsk/peg/pegininstructions/PeginInstructionsBase.java

```diff
@@ -0,0 +1,52 @@
+package co.rsk.peg.pegininstructions;
+
+import co.rsk.core.RskAddress;
+import org.ethereum.util.ByteUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+public abstract class PeginInstructionsBase implements PeginInstructions {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeginInstructionsBase.class);
+    private final int protocolVersion;
+    protected RskAddress rskDestinationAddress;
+
+    protected PeginInstructionsBase(int protocolVersion) {
+        this.protocolVersion = protocolVersion;
+    }
+
+    private RskAddress getRskDestinationAddressFromData(byte[] data) {
+        byte[] rskDestinationAddress = Arrays.copyOfRange(data, 2, 22);
+        return new RskAddress(rskDestinationAddress);
+    }
+
+    protected abstract void validateDataLength(byte[] data) throws PeginInstructionsParseException;
```

Comment:
```suggestion
    protected abstract void validateAdditionalDataLength(byte[] data) throws PeginInstructionsParseException;
```

Reply:
I think the original name makes more sense. As indeed it should validate the whole data length and not just the additional.